### PR TITLE
KAFKA-9661: Propagate includeSynonyms option to AdminClient in ConfigCommand

### PR DESCRIPTION
**Pull request by rajinisivaram**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/admin/ConfigCommand.scala [hunk 1]
```diff
@@ -367,7 +367,7 @@ object ConfigCommand extends Config {
       println(s"Completed updating default config for $entityType in the cluster.")
   }
 
-  private def describeConfig(adminClient: Admin, opts: ConfigCommandOptions): Unit = {
+  private[admin] def describeConfig(adminClient: Admin, opts: ConfigCommandOptions): Unit = {
```
Comment (mumrah) [inline core/src/main/scala/kafka/admin/ConfigCommand.scala:370]: Is this visibility just for testing?

File: core/src/main/scala/kafka/admin/ConfigCommand.scala [hunk 2]
```diff
@@ -426,7 +426,9 @@ object ConfigCommand extends Config {
       dynamicConfigSource
 
     val configResource = new ConfigResource(configResourceType, entityName)
-    val configs = adminClient.describeConfigs(Collections.singleton(configResource)).all.get(30, TimeUnit.SECONDS)
+    val describeOptions = new DescribeConfigsOptions().includeSynonyms(includeSynonyms)
```
Comment (mumrah) [inline core/src/main/scala/kafka/admin/ConfigCommand.scala:429]: I was going to ask about unused linter warnings to catch bugs like this, but turns out we have a problem in our build. Filed https://issues.apache.org/jira/browse/KAFKA-9665

**Review by mumrah — APPROVED** [review]
LGTM

File: core/src/main/scala/kafka/admin/ConfigCommand.scala (hunk 1 above)
Comment (rajinisivaram) [reply core/src/main/scala/kafka/admin/ConfigCommand.scala:370]: yes, the alterConfigs method above this had the same visibility too for the same reason.

**Comment by cmccabe** [comment]
LGTM.  Thanks, @rajinisivaram

**Pull request merged**